spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRowLevelOperationsWithLineage.java

```diff
@@ -91,7 +91,6 @@ public void beforeEach() {
     assumeThat(formatVersion).isGreaterThanOrEqualTo(3);
     // ToDo: Remove these as row lineage inheritance gets implemented in the other readers
     assumeThat(fileFormat).isEqualTo(FileFormat.PARQUET);
-    assumeThat(vectorized).isFalse();
```

Comment:
I should probably have an additional test which creates a lot of records (more than a single batch in a vectorized read), performs a modification on records with even IDs and asserts the row lineage state. Right now we're effectively just testing a single batch within this test. Theoretically everything should just work but it'll be better to have an explicit test.

Reply:
+1

Reply:
I added a test which tests a lot of records (25000, it takes 6 seconds to run locally). I didn't do the "only update even" records approach since it made the merge a bit opaque to understand. I followed what I already did in the previous test which updates and inserts a single record. The benefit now is we can essentially make sure that reading multiple batches with all the inheritance logic works and there's no weird reuse bugs.